### ISSUE TITLE
Two issues on macOS

### ISSUE THREAD

**Issue by golddranks** [flamegraph-rs/flamegraph#72]
# Outline
* `WARNING: building without debuginfo.` even though debuginfo is enabled.
* command line arguments meant for the program are passed to dtrace, fails with `dtrace: invalid probe specifier profile-997 /pid == $target/ { @[ustack(100)] = count(); }: extraneous argument 'me' ($1 is not referenced)`

# How to reproduce
## The first one
* on macOS
* `cargo install flamegraph`
* `cargo new flamegraph_test; cd flamegraph_test`
* Add to `Cargo.toml`
```
[profile.release]
debug = true
```
* Ensure debug symbols exist: `cargo build --release; ls target/release/flamegraph_test.dSYM/`
* `sudo cargo flamegraph` → see warning even if debug symbols exist

Btw. flamegraph.svg contains the function names, so apparently it does find the debug symbols, and the warning a false alarm.

## The second one
* `echo 'fn main() { println!("Hello, world! By: {:?}", std::env::args().nth(1)); }' > src/main.rs`
* `cargo run -- me`
* Prints out:
```
   Compiling flamegraph_test v0.1.0 (/Users/um003415/repos/flamegraph_test)
    Finished dev [unoptimized + debuginfo] target(s) in 3.92s
     Running `target/debug/flamegraph_test me`
Hello, world! By: Some("me")
```
* `sudo cargo flamegraph -- me`
* Prints out:
```
dtrace: invalid probe specifier profile-997 /pid == $target/ { @[ustack(100)] = count(); }: extraneous argument 'me' ($1 is not referenced)
Hello, world! By: None
failed to sample program
```

**Comment by capnfabs** [comment]
I'm also seeing both issues. I'm running in a workspace, and my code is here: https://github.com/capnfabs/blitz/tree/85fa94fe29c713d38c1ab2a4b214bc3b3fd7a153

**Comment by J-F-Liu** [comment]
me too.
```
cargo flamegraph --bin my-command -- --my-arg my-value -m -f
```
dtrace: invalid probe specifier

**Comment by hdlj** [comment]
I have the same two issues on MacOS Mojave (10.14.6) using `v0.2.0`. 
After installing the version `0.1.13`, the argument passing to the binary is working

**Comment by anweiss** [comment]
Also running into these issues

**Comment by segeljakt** [comment]
Same, on Ubuntu 18

**Comment by bjorn3** [comment]
@segeljakt What is the exact error? On Linux it shouldn't use dtrace at all.

**Comment by segeljakt** [comment]
```
16:33:13 cargo flamegraph --bin arc-script -- code "1+1+1+1"
   Compiling arc-script v0.1.0 (/home/klas/Workspace/arc-script)

    Finished release [optimized + debuginfo] target(s) in 3.97s

WARNING: building without debuginfo. Enable symbol information by adding the following lines to Cargo.toml:

[profile.release]
debug = true

Couldn't record kernel reference relocation symbol
Symbol resolution may be skewed if relocation was used (e.g. kexec).
Check /proc/kallsyms permission or run as root.

    x0: i32 = 1
    x1: i32 = 1
    x2: i32 = x0 + x1
    x3: i32 = 1
    x4: i32 = x2 + x3
    x5: i32 = 1
    x6: i32 = x4 + x5
    x6

[ perf record: Woken up 1 times to write data ]
[kernel.kallsyms] with build id 9f84fb7197d85bf2ecec720ea9188b201795a943 not found, continuing without symbols
[ perf record: Captured and wrote 0.055 MB perf.data (6 samples) ]
writing flamegraph to "flamegraph.svg"
```

**Comment by anderejd** [comment]
I have the same errors as described in the first post in this issue.

**Comment by DED8IRD** [comment]
The suggested workaround for passing arguments here https://github.com/flamegraph-rs/flamegraph/issues/70#issuecomment-598566815 worked for me

**Comment by djc** [comment]
See #88 and the follow-up in #100.

**Comment by mitsuhiko** [comment]
I believe this can be closed now as #100 was merged.

**Comment by djc** [comment]
Thanks!
